packages/react-components/react-button/library/src/components/SplitButton/useSplitButton.ts

```diff
@@ -28,14 +28,19 @@ export const useSplitButton_unstable = (
   } = props;
   const baseId = useId('splitButton-');
 
+  let menuButtonSize = size;
+  if (size === 'medium' && (appearance === 'transparent' || appearance === 'subtle')) {
+    menuButtonSize = 'large';
+  }
```

Comment:
I think this might be better put as part of only the styles and not in the state here.